packages/mui-material/src/Dialog/Dialog.d.ts

```diff
@@ -1,12 +1,96 @@
 import * as React from 'react';
 import { SxProps, Breakpoint } from '@mui/system';
-import { InternalStandardProps as StandardProps, Theme } from '..';
+import { BackdropProps, InternalStandardProps as StandardProps, Theme } from '..';
 import { PaperProps } from '../Paper';
 import { ModalProps } from '../Modal';
 import { TransitionProps } from '../transitions/transition';
 import { DialogClasses } from './dialogClasses';
+import { CreateSlotsAndSlotProps, SlotProps } from '../utils/types';
 
-export interface DialogProps extends StandardProps<ModalProps, 'children'> {
+export interface DialogSlots {
+  /**
+   * The component that renders the transition.
+   * [Follow this guide](/material-ui/transitions/#transitioncomponent-prop) to learn more about the requirements for this component.
+   * @default Collapse
+   */
+  transition?: React.JSXElementConstructor<
+    TransitionProps & { children?: React.ReactElement<any, any> }
+  >;
```

Comment:
```suggestion
  transition?: React.ElementType
```

Is there a blocker using `React.ElementType`? The reason to use it is to widening the type so user can pass their custom component.

Reply:
not really, i used previous type. updated here https://github.com/mui/material-ui/pull/44792/commits/6ae70b725e42b57a7eb55cdd24ff08e786bf641e